beanstalk.go

```diff
@@ -95,12 +96,19 @@ func ValidURIs(uris []string) error {
 	}
 
 	for _, uri := range uris {
-		hostport, _, err := parseURI(uri)
+		address, uriType, err := parseURI(uri)
 		if err != nil {
 			return err
 		}
 
-		host, _, err := net.SplitHostPort(hostport)
+		if uriType == uriUDSType {
+			if _, err := os.Stat(address); err != nil {
```

Comment:
I'm not sure, but I wonder if we should do this here. I'd imagine `ValidURIs()` would only validate the URIs and not really check if they're going to be successful when actually being used.

For example, a consumer could just be trying to reconnect to the server's domain socket by trying to open the domain socket every X amount of time.

That said, we do perform a `net.LookupHost()` call somewhere below that's in a similar vein, but a hostname not existing has less chance of resolving itself (pun) than a domain socket not being there.

What do you think?

Reply:
I was trying to follow `net.LookupHost()`. Let's keep this simple, agreed.